docs/architecture/decisions/0011-excel-generation-validation.md

```diff
@@ -0,0 +1,46 @@
+3. Consider the complexity of current excel templates / validation when making a recommendation
+4. Estimate effort various approaches
+5. (as a team) Compare estimates with user needs and make a technology decision for MVP
+
+## Decisions
+
+1. We believe sticking with one library to do everything (for the frontend and/or backend) will reduce the overhead needed to learn and work with Excel files. We also think JSON Schema is a good approach to storing requirements for the data needed from Excel files to pass between frontend and backend.
+2. We believe [ExcelJS](https://www.npmjs.com/package/exceljs) to be the best JS library to work with Excel files. It offers the ability to open and save CSV and XLSX files, work with cell-level validation, conditional formatting, cell formatting, and has easily readable and usable code. It is also open source and has most recent activity merged in from Nov 2021.
+3. After reviewing some of the generated templates used on the old FAC system, we believe the cell validations they use are rather simplistic, and are easily carried over into the newer FAC. Also taking a template and customizing it by adding user/organization information to it is a simple process with ExcelJS. Therefore, integrating this type of Excel validation should be a low hurdle to be implemented on the new system.
+4. We are unable to really estimate various approaches right now as much of _how_ the FAC will be filled out, will be stored, and will be processed on the backend hasn't been determined yet. However, we believe once those have started to be determined, it should be a reasonably small burden to add Excel validation and/or file generation to those workflows.
+5. We have not met as a team to estimate user needs yet.
+
+
+## Consequences
+
+1. We need to determine how the user will (from the frontend) fill out the forms and where they would be able to get one of these Excel templates. That will determine how and where the code to do the generation would be and how it'd integrate.
+2. We need to determine how the system will save data on the backend. This will determine what kind of data we need to get from the Excel file and save to the backend. 
+3. Depending on how we decide to move forward regarding UEI status and associated data (received from Sam.gov, manually typed in, etc.) may have an impact on how the files are generated.
+4. Depending on how this is implemented along with the rest of the FAC workflow, it may require some user research to determine the best way to do it.
+5. We could reuse Census' old template, or we could make our own. If we make our own, it may also involve user research and/or policy revisions to make sure it's adequate for the needs of all users involved (like auditors, auditees, processors, etc.)
+6. Given the flow of processing Excel files on the frontend and sending the important data to the backend, Excel files won't ever get saved. This means the concerns of the File Upload ADR won't apply here.
```

Comment:
Are there policy/legal implications to not storing the original document as a source of truth? Can the team make that decision alone, or should it be a question to OMB/the steering committee?

Reply:
Maybe a good ? for our Census friends on Thursday's meeting.

Reply:
Maybe a good ? for our Census friends on Thursday's meeting. Unless @tadhg-ohiggins knows?

Reply:
I think it's a question for Census; I feel like they've mentioned it in the past but I can't recall whether they save, for instance, the last version of Excel submitted or every version, although I think the latter is unlikely.
Since they have full bidirectional translation I think the representation of the data in their system is the canonical submitted “document”, not any Excel file that might have been part of that process.

Reply:
In any case I do not think we should regard any Excel submissions as “documents” in a formal sense, just as we would not regard a curl `POST` body as a document in a formal sense. The “documents” are the contents of the SAC as represented in our system, and the contents of the submitted PDF. Anything else is ephemeral.

Reply:
Added the following question to tomorrow's [Census agenda](https://docs.google.com/document/d/1DxuyVOusBIIRrXCaOmmzQp-Q54bthuSirmwR4j4jTbc/edit#): 

What is the current process for storing user-uploaded excel templates? Do you currently store these?

Reply:
Census stated that they do not store user-uploaded Excel fiiles.